storagenode/piecestore/endpoint.go

```diff
@@ -59,11 +59,56 @@ type Config struct {
 	MaxConcurrentRequests int           `help:"how many concurrent requests are allowed, before uploads are rejected." default:"6"`
 	OrderLimitGracePeriod time.Duration `help:"how long after OrderLimit creation date are OrderLimits no longer accepted" default:"1h0m0s"`
 	RetainTimeBuffer      time.Duration `help:"allows for small differences in the satellite and storagenode clocks" default:"1h0m0s"`
+	RetainStatus          RetainStatus  `help:"allows configuration to enable, disable, or test retain requests from the satellite" default:"disabled"`
 
 	Monitor monitor.Config
 	Sender  orders.SenderConfig
 }
 
+// RetainStatus is a type defining the enabled/disabled status of retain requests
+type RetainStatus int
+
+const (
+	// RetainDisabled means we do not do anything with retain requests
+	RetainDisabled RetainStatus = iota
```

Comment:
I would start the value with 1 because if during a development somebody forget to set the value then it will result on an error due to the zero value of an `int` is 0

Reply:
Could you clarify? I am not sure how someone could forget to set the value since there is a default set. If it is set to an invalid value, the user will get an error at the cli because of `func (v *RetainStatus) Set(s string) error {`

Reply:
I also don't understand how starting at 1 instead of 0 fixes any issues

Reply:
for example

```golang
var rstatus RetainStatus

// Do whatever logic here and the rstatus should be set to some specific
// value, if the value doesn't get set because of a bug, then the value
// will have a non valid value.

```

I prefer to have specific values rather than defaults, except when they are for end users (e.g. CLI, etc.).

When I'm reading code, if I see that var declaration and it isn't set to any value for knowing the value I have to find out which is the default while if the value has to be always set the code is easy to read.

By the way, this is a suggestion, not claiming to do this for approving it.

Reply:
Ok. The main reason I did the `iota` thing is because that seems standard for enums in golang. Do you think it would be better to just do this?
```
const (
	// RetainDisabled means we do not do anything with retain requests
	RetainDisabled = 1
	// RetainEnabled means we fully enable retain requests and delete data not defined by bloom filter
	RetainEnabled = 2
	// RetainDebug means we partially enable retain requests, and print out pieces we should delete, without actually deleting them
	RetainDebug = 3
)
```

Reply:
you can use iota and start with different values
E.g. https://play.golang.org/p/S1GQPCt16YR

Reply:
Ok, that makes sense. Updated.